examples/dev_example/docs/build.jl

```diff
@@ -0,0 +1,28 @@
+import Lexicon.Elements: config, section, document, page
+
+import Lexicon.Render: save, markdown
+
+import Docile: Cache, Collector
+
+out = document("TestDocumentation",
+    section("Introduction",
+        page("index", "index.md"),
+        ),
+    section("Docile-API",
+        page("Cache", config(style_obj="###"),
+            "# Cache",
+            "## Macros",
+            "Some intermediate text describing more about macros.",
+            (Cache, config(objfilter=[:macro])),
+            "## Methods",
+            (Cache, config(objfilter=[:method])),
+            ),
+        page("Collector", config(objfilter=[:method, :macro], style_obj="###"),
+            "# Collector",
+            "## Methods & Macros",
+            Collector,
+            ),
+        ),
+    )
```

Comment:
> Personlly I would not suggest to allow ``sections`` within a page: reason: as you said yourself:
> ```
> section would be used to separate the documentation into different parts
> ```

Using ``section`` nodes in ``page`` nodes to separate how different docstrings are displayed within a page seems to me to match up with what I said previously.

> adding additional section within a page would destroy that feature

I hope it won't :). ``section`` nodes nested in ``page`` nodes shouldn't have an effect on the folder structure. Given we're storing a ``parent`` reference it's easy to check whether a ``section`` is nested inside a ``page`` I think and handle accordingly.

> Additionally it seems to introduce even more unnecessary nesting.

``objs(...)`` is already nesting is it not? Switching to ``section(...)`` shouldn't make a difference there really.

**On the other hand:** Potentially we could rename ``objs`` to either ``objects`` or ``entries`` or ``docstrings``. I think that would be a better naming if we can't use ``section`` instead. How's that sound?

> ``document, section, page`` first argument is it's name.

Could we use a ``Symbol`` for the name then? To make it visually distinct from the text arguments. Does it always need to be specified, or can we auto-generate it if not provided?